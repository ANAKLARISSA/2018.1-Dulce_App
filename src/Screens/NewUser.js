import React from 'react';
import {View,Alert} from 'react-native';
import ScreenHeader from '../Components/ScreenHeader';
import t from 'tcomb-form-native';
import {Container} from 'native-base';
import SignUpButton from '../Components/SignUpButton';
import {newUser as styles} from './styles' ;
import axios from 'axios';

const default_state = {
  primeiroNome: '',
  segundoNome: '',
  email: '',
  senha: '',
  confirmarSenha: ''
};
const default_field_options = {
  primeiroNome: {
    error: 'Campo obrigatório'
  },
  segundoNome: {
    error: 'Campo obrigatório'
  },
  email: {
    error: 'Campo obrigatório'
  },
  senha: {
    secureTextEntry: true,
    error: 'Campo obrigatório'
  },
  confirmarSenha: {
    secureTextEntry: true,
    error: 'Campo obrigatório'
  }
};
const Form = t.form.Form;

class NewUser extends React.Component {
  constructor(props) {
    super(props);
    this.NewUser = t.struct({
      primeiroNome: t.String,
      segundoNome: t.String,
      email: t.String,
      senha: t.String,
      confirmarSenha: t.String
    });
    this.state = {
      value: default_state
    };
    this.options = {
      fields: default_field_options,
      stylesheet: formStyles
    };
  }

  onChange(value) {
    this.setState({value});
  }

  _onPress() {
    if (this.state.value.primeiroNome !== '' && this.state.value.segundoNome !== '' && this.state.value.email !== '') {
      if (this.state.value.senha === this.state.value.confirmarSenha) {
        const url = 'http://52.67.4.137:8083/api/user/create';
        axios.post(url, {
          firstName: this.state.value.primeiroNome,
          lastName: this.state.value.segundoNome,
          email: this.state.value.email,
          password: this.state.value.senha
        })
          .then((response) => {
            var error = '';
            if (response.data.email_valid_error) {
              error += response.data.email_valid_error;
            }
            if (response.data.email_duplicate_error) {
              error += response.data.email_duplicate_error;
            }
            if (error !== '') {
              Alert.alert('Erro!', error);
            } else {
              this.props.navigation.navigate('login');
              Alert.alert('Cadastro feito com sucesso!');
            }
          })
          .catch((err) => {
            Alert.alert('Erro!', err.response.data.message);
          });
      }
      else {
        Alert.alert('As senhas devem ser iguais!');
      }
    } else {
      Alert.alert('Os campos não podem estar vazios!');
    }
  }

  render() {

    return (
      <Container style={{backgroundColor: '#FFF'}}>
        <ScreenHeader title = 'Criar nova conta'/>
        <View style={styles.container}>
          <Form
            ref='form'
            type={this.NewUser}
            value={this.state.value}
            options={this.options}
            onChange={(v) => this.onChange(v)}
          />
        </View>
        <View style={{flex: 1}}>
          <SignUpButton
            text = 'Cadastrar'
            onPress = {(v) => this._onPress(v)}
          />
        </View>
      </Container>
    );
  }
}

const formStyles = {
  ...Form.stylesheet,
  error: {
    color: 'red',
    fontSize: 14,
    marginBottom: 3,
    fontWeight: '600'
  }
};

export default NewUser;
