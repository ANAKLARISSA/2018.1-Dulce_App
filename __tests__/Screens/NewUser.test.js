import React from 'react';
import 'react-native';
import NewUser from '../../src/Screens/NewUser';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
import {shallow} from 'enzyme';
import t from 'tcomb-form-native';

require('bezier');
Enzyme.configure({adapter: new Adapter()});

it('renders correctly', () => {
  const tree = renderer
    .create(<NewUser />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('test onChange value',() => {
  const wrapper = shallow(<NewUser/>);
  const onChange = wrapper.instance().onChange();
  console.log(onChange);
});

it('testing onChange container',() => {
  const type = t.struct({
    nome: t.String,
    email: t.String,
    senha: t.String,
    confirmarSenha: t.String
  });
  const wrapper = shallow(<NewUser type={type}/>);
  const container = wrapper.find('Styled(Container)').at(0);
  const form = container.find('Form').at(0);
  form.props().onChange();
});

it('testing onPress Cadastro',() => {
  const spy = jest.spyOn(NewUser.prototype, '_onPress');
  const wrapper = shallow(<NewUser/>);
  const SignUp = wrapper.find('SignUp').at(0);
  SignUp.simulate('press');
  expect(spy).toHaveBeenCalled();
});

jest.mock('Alert', () => {
  return {
    alert: jest.fn()
  };
});

it('testing function _onPress',() => {
  const type = t.struct({
    nome: t.String,
    email: t.String,
    senha: t.String,
    confirmarSenha: t.String
  });
  const value = {
    nome: 'Ezequiel',
    email: '@',
    senha: '123',
    confirmarSenha: '123'
  };
  const wrapper = shallow(<NewUser type={type}/>);
  wrapper.setState({value: value});
  const onPressF = wrapper.instance()._onPress();
  console.log(wrapper.debug());
  console.log(onPressF);
});

it('testing function _onPress error',() => {
  const type = t.struct({
    nome: t.String,
    email: t.String,
    senha: t.String,
    confirmarSenha: t.String
  });
  const value = {
    nome: 'Ezequiel',
    email: '@',
    senha: '123',
    confirmarSenha: '132'
  };
  const wrapper = shallow(<NewUser type={type}/>);
  wrapper.setState({value: value});
  const onPressF = wrapper.instance()._onPress();
  console.log(wrapper.debug());
  console.log(onPressF);
});
