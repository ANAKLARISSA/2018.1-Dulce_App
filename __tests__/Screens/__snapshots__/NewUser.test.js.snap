// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "height": 1334,
      },
      Object {
        "backgroundColor": "#FFF",
      },
    ]
  }
  virtual={undefined}
>
  <View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F8F8F8",
            "borderBottomColor": "#a7a6ab",
            "borderBottomWidth": 0.5,
            "elevation": 3,
            "flexDirection": "row",
            "height": 64,
            "justifyContent": "center",
            "left": 0,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 15,
            "right": 0,
            "shadowColor": undefined,
            "shadowOffset": undefined,
            "shadowOpacity": undefined,
            "shadowRadius": undefined,
            "top": 0,
          },
          Object {
            "backgroundColor": "#5f4b8b",
          },
        ]
      }
      virtual={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "alignSelf": "center",
            "flex": 1,
          }
        }
        virtual={undefined}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": null,
              "backgroundColor": "transparent",
              "borderBottomWidth": null,
              "borderColor": null,
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "elevation": 0,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "space-between",
              "marginLeft": -3,
              "opacity": 1,
              "paddingBottom": 6,
              "paddingLeft": 6,
              "paddingRight": 6,
              "paddingTop": 6,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Array [
                  Object {
                    "color": "#000",
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#007aff",
                    "fontSize": 30,
                    "marginLeft": 2,
                    "marginRight": 2,
                    "marginTop": 0,
                    "paddingTop": 2,
                  },
                ],
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
            virtual={undefined}
          >
            Ôèè
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "flex": 1,
            },
            Object {},
          ]
        }
        virtual={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 17,
                "fontWeight": "600",
                "textAlign": "center",
              },
              Object {
                "fontFamily": "Raleway",
              },
            ]
          }
          virtual={undefined}
        >
           
          Criar nova conta
           
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#FFF",
        "justifyContent": "center",
        "marginTop": 10,
        "padding": 15,
      }
    }
  >
    <View
      style={Object {}}
    >
      <View
        style={
          Object {
            "marginBottom": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#1a1515",
              "fontFamily": "Lato",
              "fontSize": 17,
              "fontWeight": "200",
              "marginBottom": 7,
            }
          }
        >
          Primeiro nome
        </Text>
        <View
          style={Object {}}
        >
          <TextInput
            accessibilityLabel="Primeiro nome"
            allowFontScaling={true}
            autoCapitalize={undefined}
            autoCorrect={undefined}
            autoFocus={undefined}
            blurOnSubmit={undefined}
            clearButtonMode={undefined}
            clearTextOnFocus={undefined}
            editable={undefined}
            enablesReturnKeyAutomatically={undefined}
            keyboardAppearance={undefined}
            keyboardType={undefined}
            maxLength={undefined}
            multiline={undefined}
            numberOfLines={undefined}
            onBlur={undefined}
            onChange={undefined}
            onChangeText={[Function]}
            onContentSizeChange={undefined}
            onEndEditing={undefined}
            onFocus={undefined}
            onKeyPress={undefined}
            onLayout={undefined}
            onSelectionChange={undefined}
            onSubmitEditing={undefined}
            placeholder={undefined}
            placeholderTextColor={undefined}
            returnKeyType={undefined}
            secureTextEntry={undefined}
            selectTextOnFocus={undefined}
            selectionColor={undefined}
            selectionState={undefined}
            style={
              Object {
                "borderColor": "#cccccc",
                "borderRadius": 20,
                "borderWidth": 1,
                "color": "#1a1515",
                "fontSize": 17,
                "height": 36,
                "marginBottom": 5,
                "paddingHorizontal": 7,
                "paddingVertical": 7,
              }
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
      <View
        style={
          Object {
            "marginBottom": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#1a1515",
              "fontFamily": "Lato",
              "fontSize": 17,
              "fontWeight": "200",
              "marginBottom": 7,
            }
          }
        >
          Segundo nome
        </Text>
        <View
          style={Object {}}
        >
          <TextInput
            accessibilityLabel="Segundo nome"
            allowFontScaling={true}
            autoCapitalize={undefined}
            autoCorrect={undefined}
            autoFocus={undefined}
            blurOnSubmit={undefined}
            clearButtonMode={undefined}
            clearTextOnFocus={undefined}
            editable={undefined}
            enablesReturnKeyAutomatically={undefined}
            keyboardAppearance={undefined}
            keyboardType={undefined}
            maxLength={undefined}
            multiline={undefined}
            numberOfLines={undefined}
            onBlur={undefined}
            onChange={undefined}
            onChangeText={[Function]}
            onContentSizeChange={undefined}
            onEndEditing={undefined}
            onFocus={undefined}
            onKeyPress={undefined}
            onLayout={undefined}
            onSelectionChange={undefined}
            onSubmitEditing={undefined}
            placeholder={undefined}
            placeholderTextColor={undefined}
            returnKeyType={undefined}
            secureTextEntry={undefined}
            selectTextOnFocus={undefined}
            selectionColor={undefined}
            selectionState={undefined}
            style={
              Object {
                "borderColor": "#cccccc",
                "borderRadius": 20,
                "borderWidth": 1,
                "color": "#1a1515",
                "fontSize": 17,
                "height": 36,
                "marginBottom": 5,
                "paddingHorizontal": 7,
                "paddingVertical": 7,
              }
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
      <View
        style={
          Object {
            "marginBottom": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#1a1515",
              "fontFamily": "Lato",
              "fontSize": 17,
              "fontWeight": "200",
              "marginBottom": 7,
            }
          }
        >
          Email
        </Text>
        <View
          style={Object {}}
        >
          <TextInput
            accessibilityLabel="Email"
            allowFontScaling={true}
            autoCapitalize={undefined}
            autoCorrect={undefined}
            autoFocus={undefined}
            blurOnSubmit={undefined}
            clearButtonMode={undefined}
            clearTextOnFocus={undefined}
            editable={undefined}
            enablesReturnKeyAutomatically={undefined}
            keyboardAppearance={undefined}
            keyboardType={undefined}
            maxLength={undefined}
            multiline={undefined}
            numberOfLines={undefined}
            onBlur={undefined}
            onChange={undefined}
            onChangeText={[Function]}
            onContentSizeChange={undefined}
            onEndEditing={undefined}
            onFocus={undefined}
            onKeyPress={undefined}
            onLayout={undefined}
            onSelectionChange={undefined}
            onSubmitEditing={undefined}
            placeholder={undefined}
            placeholderTextColor={undefined}
            returnKeyType={undefined}
            secureTextEntry={undefined}
            selectTextOnFocus={undefined}
            selectionColor={undefined}
            selectionState={undefined}
            style={
              Object {
                "borderColor": "#cccccc",
                "borderRadius": 20,
                "borderWidth": 1,
                "color": "#1a1515",
                "fontSize": 17,
                "height": 36,
                "marginBottom": 5,
                "paddingHorizontal": 7,
                "paddingVertical": 7,
              }
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
      <View
        style={
          Object {
            "marginBottom": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#1a1515",
              "fontFamily": "Lato",
              "fontSize": 17,
              "fontWeight": "200",
              "marginBottom": 7,
            }
          }
        >
          Senha
        </Text>
        <View
          style={Object {}}
        >
          <TextInput
            accessibilityLabel="Senha"
            allowFontScaling={true}
            autoCapitalize={undefined}
            autoCorrect={undefined}
            autoFocus={undefined}
            blurOnSubmit={undefined}
            clearButtonMode={undefined}
            clearTextOnFocus={undefined}
            editable={undefined}
            enablesReturnKeyAutomatically={undefined}
            keyboardAppearance={undefined}
            keyboardType={undefined}
            maxLength={undefined}
            multiline={undefined}
            numberOfLines={undefined}
            onBlur={undefined}
            onChange={undefined}
            onChangeText={[Function]}
            onContentSizeChange={undefined}
            onEndEditing={undefined}
            onFocus={undefined}
            onKeyPress={undefined}
            onLayout={undefined}
            onSelectionChange={undefined}
            onSubmitEditing={undefined}
            placeholder={undefined}
            placeholderTextColor={undefined}
            returnKeyType={undefined}
            secureTextEntry={true}
            selectTextOnFocus={undefined}
            selectionColor={undefined}
            selectionState={undefined}
            style={
              Object {
                "borderColor": "#cccccc",
                "borderRadius": 20,
                "borderWidth": 1,
                "color": "#1a1515",
                "fontSize": 17,
                "height": 36,
                "marginBottom": 5,
                "paddingHorizontal": 7,
                "paddingVertical": 7,
              }
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
      <View
        style={
          Object {
            "marginBottom": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#1a1515",
              "fontFamily": "Lato",
              "fontSize": 17,
              "fontWeight": "200",
              "marginBottom": 7,
            }
          }
        >
          Confirmar senha
        </Text>
        <View
          style={Object {}}
        >
          <TextInput
            accessibilityLabel="Confirmar senha"
            allowFontScaling={true}
            autoCapitalize={undefined}
            autoCorrect={undefined}
            autoFocus={undefined}
            blurOnSubmit={undefined}
            clearButtonMode={undefined}
            clearTextOnFocus={undefined}
            editable={undefined}
            enablesReturnKeyAutomatically={undefined}
            keyboardAppearance={undefined}
            keyboardType={undefined}
            maxLength={undefined}
            multiline={undefined}
            numberOfLines={undefined}
            onBlur={undefined}
            onChange={undefined}
            onChangeText={[Function]}
            onContentSizeChange={undefined}
            onEndEditing={undefined}
            onFocus={undefined}
            onKeyPress={undefined}
            onLayout={undefined}
            onSelectionChange={undefined}
            onSubmitEditing={undefined}
            placeholder={undefined}
            placeholderTextColor={undefined}
            returnKeyType={undefined}
            secureTextEntry={true}
            selectTextOnFocus={undefined}
            selectionColor={undefined}
            selectionState={undefined}
            style={
              Object {
                "borderColor": "#cccccc",
                "borderRadius": 20,
                "borderWidth": 1,
                "color": "#1a1515",
                "fontSize": 17,
                "height": 36,
                "marginBottom": 5,
                "paddingHorizontal": 7,
                "paddingVertical": 7,
              }
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "height": 1334,
        }
      }
      virtual={undefined}
    >
      <RCTScrollView
        automaticallyAdjustContentInsets={false}
        contentContainerStyle={
          Array [
            Object {
              "padding": undefined,
            },
            undefined,
          ]
        }
        contentInset={
          Object {
            "bottom": 0,
          }
        }
        enableAutomaticScroll={true}
        enableResetScrollToCoords={true}
        extraHeight={75}
        extraScrollHeight={0}
        getScrollResponder={[Function]}
        handleOnScroll={[Function]}
        keyboardDismissMode="interactive"
        keyboardOpeningTime={250}
        keyboardShouldPersistTaps="handled"
        keyboardSpace={0}
        onScroll={[Function]}
        resetKeyboardSpace={[Function]}
        resetScrollToCoords={
          Object {
            "x": 0,
            "y": 0,
          }
        }
        scrollEventThrottle={1}
        scrollForExtraHeightOnAndroid={[Function]}
        scrollToEnd={[Function]}
        scrollToFocusedInput={[Function]}
        scrollToPosition={[Function]}
        showsVerticalScrollIndicator={true}
        style={
          Object {
            "backgroundColor": "transparent",
            "flex": 1,
          }
        }
        viewIsInsideTabBar={false}
        virtual={undefined}
      >
        <View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "#5f4b8b",
                "borderBottomWidth": null,
                "borderColor": "#007aff",
                "borderLeftWidth": null,
                "borderRadius": 57,
                "borderRightWidth": null,
                "borderTopWidth": null,
                "elevation": 2,
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "center",
                "marginBottom": 0,
                "marginTop": 30,
                "opacity": 1,
                "paddingBottom": 6,
                "paddingHorizontal": 50,
                "paddingTop": 6,
                "width": 180,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignItems": "center",
                  "color": "white",
                  "fontFamily": "Raleway",
                  "fontSize": 16,
                }
              }
            >
              Cadastrar
            </Text>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</View>
`;
